test/plugin/test_out_detect_exceptions.rb

```diff
@@ -98,6 +114,80 @@ def test_exception_detection
     assert_equal(make_logs(t, *messages), d.events)
   end
 
+  def test_ignore_exception_nested_in_json
```

Comment:
Can we combine these two tests? Seems the main difference is the `lines` test data input.